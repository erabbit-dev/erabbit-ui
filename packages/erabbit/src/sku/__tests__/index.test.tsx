import { Sku } from '..'
import { mount } from '@vue/test-utils'
import data from '../constants'
import { ref } from 'vue'

test('should render default sku', () => {
  const wrapper = mount({
    setup() {
      const goods = ref(data)
      return () => <Sku specs={goods.value.specs} skus={goods.value.skus}></Sku>
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
})

test('should render set props sku-id', async () => {
  const wrapper = mount({
    setup() {
      const goods = ref(data)
      const skuId = ref('300206065')
      return () => (
        <Sku
          specs={goods.value.specs}
          skus={goods.value.skus}
          skuId={skuId.value}
        ></Sku>
      )
    },
  })
  expect(wrapper.html()).toMatchSnapshot()

  expect(wrapper.findAll('.is-selected').length).toBe(2)

  await wrapper.setProps({ skuId: null })

  expect(wrapper.findAll('.is-selected').length).toBe(0)
})

test('should set sku-id when trigger change event', async () => {
  let skuId: string | undefined = '300206065'
  const wrapper = mount({
    setup() {
      const goods = ref(data)
      const onChange = (id?: string) => {
        skuId = id
      }
      return () => (
        <Sku
          specs={goods.value.specs}
          skus={goods.value.skus}
          skuId={skuId}
          onChange={(e) => onChange(e.id)}
        ></Sku>
      )
    },
  })

  await wrapper.find('.is-selected').trigger('click')

  expect(skuId).toBe(undefined)

  await wrapper
    .find('.er-sku__row:first-child .er-sku__img:first-child')
    .trigger('click')
  await wrapper
    .find('.er-sku__row:last-child .er-sku__btn:nth-last-child(2)')
    .trigger('click')
  expect(skuId).toBe('300206064')
})
