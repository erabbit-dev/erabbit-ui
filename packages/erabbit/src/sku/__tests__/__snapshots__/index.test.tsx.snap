// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render default sku 1`] = `
"<div class=\\"er-sku\\">
  <dl class=\\"er-sku__row\\">
    <dt class=\\"er-sku__label\\">颜色</dt>
    <dd class=\\"er-sku__wrapper\\"><img class=\\"er-sku__img\\" src=\\"https://yanxuan-item.nosdn.127.net/e19067221f1e4e93b2afca3973479496.png\\" alt=\\"白色（女）\\" title=\\"白色（女）\\"><img class=\\"er-sku__img\\" src=\\"https://yanxuan-item.nosdn.127.net/51e0f8bd91e0a4987a797b2586be1ec1.png\\" alt=\\"蓝白（女）\\" title=\\"蓝白（女）\\"><img class=\\"er-sku__img\\" src=\\"https://yanxuan-item.nosdn.127.net/71354df110d7a93e1c46d51fcdd5840e.png\\" alt=\\"白黑（男）\\" title=\\"白黑（男）\\"><img class=\\"er-sku__img\\" src=\\"https://yanxuan-item.nosdn.127.net/032583c92671e01c7ca417210fa29c75.png\\" alt=\\"黑色（男）\\" title=\\"黑色（男）\\"></dd>
  </dl>
  <dl class=\\"er-sku__row\\">
    <dt class=\\"er-sku__label\\">尺码</dt>
    <dd class=\\"er-sku__wrapper\\"><span class=\\"er-sku__btn\\">女S(35-36)</span><span class=\\"er-sku__btn\\">女L(39-40)</span><span class=\\"er-sku__btn\\">男M（40-41）</span><span class=\\"er-sku__btn\\">男L（42-43）</span><span class=\\"er-sku__btn\\">女M(37-38)</span><span class=\\"er-sku__btn\\">男XL(44-45)</span></dd>
  </dl>
</div>"
`;

exports[`should render set props sku-id 1`] = `
"<div class=\\"er-sku\\">
  <dl class=\\"er-sku__row\\">
    <dt class=\\"er-sku__label\\">颜色</dt>
    <dd class=\\"er-sku__wrapper\\"><img class=\\"er-sku__img is-selected\\" src=\\"https://yanxuan-item.nosdn.127.net/e19067221f1e4e93b2afca3973479496.png\\" alt=\\"白色（女）\\" title=\\"白色（女）\\"><img class=\\"er-sku__img\\" src=\\"https://yanxuan-item.nosdn.127.net/51e0f8bd91e0a4987a797b2586be1ec1.png\\" alt=\\"蓝白（女）\\" title=\\"蓝白（女）\\"><img class=\\"er-sku__img is-disabled\\" src=\\"https://yanxuan-item.nosdn.127.net/71354df110d7a93e1c46d51fcdd5840e.png\\" alt=\\"白黑（男）\\" title=\\"白黑（男）\\"><img class=\\"er-sku__img is-disabled\\" src=\\"https://yanxuan-item.nosdn.127.net/032583c92671e01c7ca417210fa29c75.png\\" alt=\\"黑色（男）\\" title=\\"黑色（男）\\"></dd>
  </dl>
  <dl class=\\"er-sku__row\\">
    <dt class=\\"er-sku__label\\">尺码</dt>
    <dd class=\\"er-sku__wrapper\\"><span class=\\"er-sku__btn\\">女S(35-36)</span><span class=\\"er-sku__btn is-selected\\">女L(39-40)</span><span class=\\"er-sku__btn is-disabled\\">男M（40-41）</span><span class=\\"er-sku__btn is-disabled\\">男L（42-43）</span><span class=\\"er-sku__btn\\">女M(37-38)</span><span class=\\"er-sku__btn is-disabled\\">男XL(44-45)</span></dd>
  </dl>
</div>"
`;
